extra_space.py

```diff
@@ -0,0 +1,51 @@
+# !/usr/bin/env python
+'''
+This module automatically checks for trailing spaces.
+It scans the complete folder, searches for python files
+(ignoring the virtual environment python files) and automatially
+removes the trailing spaces.
+'''
+
+import os
+import sys
+
+
+BASE_DIR = os.path.abspath(__file__)
+
+
+def fix_file(path):
+    '''
+    The function, using 'for' loop scans all the directories as well
+    as sub-directories of the folder for python file and checks
+    for trailing white space.
+    '''
+    p = ""
+    for dirpath, _, filenames in os.walk(path):
+        if(os.path.exists(os.path.join(dirpath, 'bin/activate'))):
+            p = dirpath
+        for f in filenames:
+            if f.endswith('.py'):
+                filename = os.path.abspath(os.path.join(dirpath, f))
+                if (not bool(p)):
+                    if os.access(filename, os.W_OK):
+                        try:
+                            file1 = open(filename, 'r')
+                            lines = file1.readlines()
+                            for index, line in enumerate(lines):
+                                if line.endswith(" \n"):
+                                    lines[index] = line.rstrip() + "\n"
+                                if line.endswith("\r\n"):
+                                    lines[index] = line.rstrip() + "\n"
```

Comment:
Why not do `lines[index] = line.rstrip() + "\n"` without any checks? The outcome will be the same as `.strip()` on a line without spaces at the end simply won't do anything (other than removing the `\n` which you're adding again)